### ISSUE TITLE
`cachix: CppStdException "Exception: path '/nix/store/...' is not valid; type: nix::InvalidPath"`

### ISSUE THREAD

**Issue by samuela** [cachix/cachix-action#106]
I'm seeing a weird error trying to use cachix-action. 
```
/home/runner/work/_actions/cachix/cachix-action/v10/dist/main/push-paths.sh /home/runner/.nix-profile/bin/cachix ploop 
cachix: CppStdException "Exception: path '/nix/store/ab3yh7s8rms16b66jlbbzvjaif94ls93-python3.9-apache-beam-2.35.0' is not valid; type: nix::InvalidPath"
Error: Action failed with error: Error: The process '/home/runner/work/_actions/cachix/cachix-action/v10/dist/main/push-paths.sh' failed with exit code 1
```

Here's a link to the failed run: https://github.com/samuela/nixpkgs-upkeep/runs/4976168659?check_suite_focus=true.

Am I doing something wrong?

**Comment by samuela** [comment]
Just got the same thing on a different job:
```
cachix: CppStdException "Exception: path '/nix/store/qli5gyjqm0lizmxz2lqmybwgd3pvz1ji-tensorflow-gpu-2.7.0.drv.chroot' is not valid; type: nix::InvalidPath"
Error: Action failed with error: Error: The process '/home/runner/work/_actions/cachix/cachix-action/v10/dist/main/push-paths.sh' failed with exit code 1
```
Here's the [run](https://github.com/samuela/nixpkgs-upkeep/runs/5012733644?check_suite_focus=true).

**Comment by samuela** [comment]
EDIT: in both of these cases I'm realizing that there were build errors in the job... My expectation was that nix would leave partial file junk in /nix/store though?

**Comment by steveeJ** [comment]
i was wondering how to install cachix 0.7.0 using `cachix/cachix-action@v10` today, and came up with the following solution:

```yml
- name: Setup cachix
  uses: cachix/cachix-action@v10
  with:
    (...)
    installCommand: |
      nix-env -if https://github.com/cachix/cachix/tarball/cachix-api-0.7.0 \
        --substituters 'https://cache.nixos.org https://cachix.cachix.org' \
        --trusted-public-keys 'cachix.cachix.org-1:eWNHQldwUO7G2VkjpnjDbWwy4KQ/HNxht7H4SSoMckM= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY='
```

**Comment by steveeJ** [comment]
installing cachix 0.7 as shown above doesn't solve my instance of the issue.
error copied [from this run](https://github.com/holochain/holochain/runs/5234764519?check_suite_focus=true#step:25:4):

```
Cachix: push
  /Users/runner/work/_actions/cachix/cachix-action/v10/dist/main/push-paths.sh /Users/runner/.nix-profile/bin/cachix holochain-ci 
  cachix: CppStdException "Exception: \ESC[31;1merror:\ESC[0m path '\ESC[35;1m/nix/store/9lyllc4w3il5kkkmfnla455min34yz7b-rust-overlay-b4532ea\ESC[0m' is not valid; type: nix::InvalidPath"
  Error: Action failed with error: Error: The process '/Users/runner/work/_actions/cachix/cachix-action/v10/dist/main/push-paths.sh' failed with exit code 1
```

**Comment by domenkozar** [comment]
I've released 0.7.0 and it should be the default, does that work now?

**Comment by austinbutler** [comment]
Tried again today with `cachix-action@10`, same issue.

**Comment by domenkozar** [comment]
I'll look into this.

**Comment by flokli** [comment]
I can also see this happening. I opened https://github.com/cachix/cachix-action/issues/111, which is related.

**Comment by purefn** [comment]
Anyone know of a workaround for this issue? I'm running into it on a GitHub action that ends up building GHC. It eventually times out. Then it hits this error and so every build needs to rebuild GHC and ends up timing out and failing.

**Comment by domenkozar** [comment]
That's terrible experience. I'll prioritize this and look into it tomorrow!

**Comment by domenkozar** [comment]
To reproduce:

```
nix-build -E '(import <nixpkgs> {}).runCommand "foo" {} "sleep 999999999"'
````

Then in another terminal:
```
$ find /nix/store -maxdepth 1 -name *.drv.chroot
/nix/store/nz0ziflbx07n04szdhg0f60jk5givdla-foo.drv.chroot
```

And finally:

```
$ cachix push mycache /nix/store/nz0ziflbx07n04szdhg0f60jk5givdla-foo.drv.chroot
cachix: CppStdException e "\ESC[31;1merror:\ESC[0m path '\ESC[35;1m/nix/store/nz0ziflbx07n04szdhg0f60jk5givdla-foo.drv.chroot\ESC[0m' is not valid"(Just "nix::InvalidPath")
``

**Comment by domenkozar** [comment]
This is now fixed in https://github.com/cachix/cachix/pull/410

I'll make a release and let you know once it hits the channels!

Thanks for reporting!

**Comment by domenkozar** [comment]
I've released 0.7.1, once it reaches nixpkgs I'll publish it as official (hopefully in a few days).

**Comment by domenkozar** [comment]
Tracking at https://github.com/NixOS/nixpkgs/pull/179211
